CRM/Event/Form/Registration/Confirm.php

```diff
@@ -1052,6 +1052,9 @@ public static function fixLocationFields(&$params, &$fields, &$form) {
       }
```

Comment:
This function is largely a duplicate of the contribution function    

$fields = $this->formatParamsForPaymentProcessor($fields);

However, there is a possible case where the syntax might diverge when dealing with multiple participants.

In general the contribution code is cleaner because rather than coming up with 100  & one conditions as to why we add or remove the billing fields it simply assumes that if we presented them on the form in the first place then we probably did so in order to save them